client/my-sites/plans-features-main/hooks/use-plan-upgrade-credits.ts

```diff
@@ -0,0 +1,40 @@
+import { useSelector } from 'react-redux';
+import { getPlanDiscountedRawPrice } from 'calypso/state/sites/plans/selectors';
+import { getSitePlanRawPrice } from 'calypso/state/sites/plans/selectors/get-site-plan-raw-price';
+import isPlanAvailableForPurchase from 'calypso/state/sites/plans/selectors/is-plan-available-for-purchase';
+import type { PlanSlug } from '@automattic/calypso-products';
+
+/**
+ * Calculate available plan credits given a set of displayed plans
```

Comment:
NIT - I think it might help if we are a little more specific here on how these credits are derived. Initially, I thought this will do some sort of summation, but it picks the max credits resulting from any plan.